### ISSUE TITLE
Add UI for adding work IDs

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#3430]
Currently we have no way to store identifiers on a work. Loads of identifier (Wikidata IDs, GoodReads work IDs, LibraryThing IDs) apply only to works, and not editions.

### Summary of Requirements
- [x] Should follow the same JSON schema as identifier on editions (e.g. `"identifiers": {"goodreads": ["6675490"], "wikidata": ["Q91804864"], "librarything": ["9387249"]}`)
- [x] Should re-use the same code used on editions page

### Additional context
- Blocks: #497
- Blocks: #9798

### Stakeholders
@seabelis

**Comment by BrittanyBunk** [comment]
Blocks: #3433. Until the UI for works and editions are separated out, it'll be too difficult and confusing for members to add work IDs when editing.

**Comment by seabelis** [comment]
> Loads of identifier (Wikidata IDs, GoodReads work IDs, LibraryThing IDs) apply only to works, and not editions.

Goodreads IDs are edition specific. These should remain at the edition-level.

**Comment by cdrini** [comment]
👍 The majority should remain on the edition. But note that there are GoodReads work-specific IDs, they're just hard to find (e.g. https://www.goodreads.com/work/editions/2776383-the-driver-s-seat <- this url contains the work id). We might have to have some warning to avoid this being used incorrectly though, since I imagine it would be confusing.

**Comment by seabelis** [comment]
Good point. I hadn't considered that as you have to get there via an edition. But it's subject to change. Because of the way the GR combination and separation works, it's possible if this gets combined incorrectly with a work that has an edition that has more "shelvings" than other editions, the work will then become that other work and the work with fewer editions will then have to be separated out into a new work. I'll see if I can find an example.

**Comment by RayBB** [comment]
Pretty sure this is a duplicate of #1797

**Comment by LeadSongDog** [comment]
@RayBB Not quite, that #1797 is just Wikidata IDs, but there are multiple work ID types just as there are multiple edition IDs. There are also multiple work classification IDs (see #897) for which we will still need to provide a distinct UI.

**Comment by tfmorris** [comment]
> That OCAID are Edition level identifiers, and similarly to above, special code to handle them can be deleted.

Internet Archive IDs don't actually correspond 1:1 with editions, but rather identify a particular scan/volume. Multi-volume editions and editions which have been scanned multiple times with both have multiple IA identifiers.

**Comment by daniel-barrows** [comment]
Just a note since some of the above comments seems confused. Wikidata, GoodReads, LibraryThing, Open Library all have identifiers for both works and editions. GoodReads tends to show users edition pages, whereas LibraryThing tends to show users work pages, but all have both.

**Comment by schu96** [comment]
I'd like to be assigned to this issue

**Comment by daniel-barrows** [comment]
@seabelis GoodReads has both edition and work IDs
Work ID example: https://www.goodreads.com/work/editions/4640799
Edition ID example: https://www.goodreads.com/book/show/3.Harry_Potter_and_the_Sorcerer_s_Stone

**Comment by daniel-barrows** [comment]
@cdrini Wikidata has items for editions as well as works. Wikidata currently has about [590,436 editions](https://query.wikidata.org/#SELECT%20%28COUNT%28%3Fitem%29%20AS%20%3Fcount%29%7B%0A%20%20%7B%0A%20%20%20%20%3Fitem%20wdt%3AP31%20wd%3AQ3331189.%0A%20%20%7D%20UNION%20%7B%0A%20%20%20%20%3Fitem%20wdt%3AP31%20%3Fclass.%0A%20%20%20%20%3Fclass%20wdt%3AP279%20wd%3AQ3331189.%0A%20%20%7D%0A%7D), and [1,356,909 works](https://query.wikidata.org/#SELECT%20%28COUNT%28%3Fitem%29%20AS%20%3Fcount%29%7B%0A%20%20%7B%0A%20%20%20%20%3Fitem%20wdt%3AP31%20wd%3AQ7725634.%0A%20%20%7D%20UNION%20%7B%0A%20%20%20%20%3Fitem%20wdt%3AP31%20wd%3AQ47461344.%0A%20%20%7D%20UNION%20%7B%0A%20%20%20%20%3Fitem%20wdt%3AP31%20%3Fclass.%0A%20%20%20%20%3Fclass%20wdt%3AP279%20wd%3AQ7725634.%0A%20%20%7D%20UNION%20%7B%0A%20%20%20%20%3Fitem%20wdt%3AP31%20%3Fclass.%0A%20%20%20%20%3Fclass%20wdt%3AP279%20wd%3AQ47461344.%0A%20%20%7D%0A%7D). As an identifier, Wikidata item IDs can be both work and edition identifiers.
